theseus_gui/src/pages/Settings.vue

```diff
@@ -1,7 +1,319 @@
+import { get, set, deepEqual } from '@/helpers/settings'
+
+// Original object to track changes
+const originalSettings = ref(await get())
+// Object to bind
+const settings = ref(await get())
+
+const saveButton = ref(null)
+
+if (!settings.value.java_globals.JAVA_8)
+  settings.value.java_globals.JAVA_8 = { path: '', version: '' }
+if (!settings.value.java_globals.JAVA_17)
+  settings.value.java_globals.JAVA_17 = { path: '', version: '' }
+
+watch(settings.value, (newSettings) => {
+  // Validate the changed state
+  if (newSettings.hooks.pre_launch === '') delete settings.value.hooks.pre_launch
+  if (newSettings.hooks.wrapper === '') delete settings.value.hooks.wrapper
+  if (newSettings.hooks.post_exit === '') delete settings.value.hooks.post_exit
+
+  settings.value.max_concurrent_downloads = parseInt(newSettings.max_concurrent_downloads)
+
+  if (deepEqual(originalSettings.value, settings.value)) saveButton.value.$el.style.opacity = 0
+  else saveButton.value.$el.style.opacity = 1
+
+  if (!settings.value.java_globals.JAVA_8)
+    settings.value.java_globals.JAVA_8 = { path: '', version: '' }
+  if (!settings.value.java_globals.JAVA_17)
+    settings.value.java_globals.JAVA_17 = { path: '', version: '' }
+})
+
+const themeStore = useTheming()
+
+const handleTheme = (e) => themeStore.setThemeState(e.option.toLowerCase())
+const saveJavaPath = () => {}
+const saveSettings = async () => {
+  await set(settings.value)
+  saveButton.value.$el.style.opacity = 0
+}
+</script>
 
 <template>
   <div>
-    <p>Settings page</p>
+    <Button ref="saveButton" color="primary" class="save-btn" @click="saveSettings">Save</Button>
+    <Card class="theming">
+      <h2>Themes</h2>
+      <div class="setting-row">
+        <div class="description">
+          <h3>Color theme</h3>
+          <p>Change the global launcher color theme.</p>
+        </div>
+        <DropdownSelect
+          name="Theme dropdown"
+          :options="themeStore.themeOptions"
+          :default-value="themeStore.selectedTheme"
+          :model-value="themeStore.selectedTheme"
+          class="theme-dropdown"
+          @change="handleTheme"
+        />
+      </div>
+    </Card>
+    <Card class="settings-card">
+      <h2 class="settings-title">Java</h2>
+      <div class="settings-group">
+        <h3>Java 17 Location</h3>
+        <div class="toggle-setting">
+          <input
+            v-model="settings.java_globals.JAVA_17.path"
+            type="text"
+            class="input installation-input"
+            placeholder="/path/to/java17"
+          />
+          <span class="installation-buttons">
+            <Button @click="saveJavaPath">
+              <SearchIcon />
+              Auto Detect
+            </Button>
+            <Button @click="saveJavaPath">
+              <BrowseIcon />
+              Browse
+            </Button>
+            <Button @click="saveJavaPath">
+              <PlayIcon />
+              Test
+            </Button>
```

Comment:
these buttons don't work. the auto detect should launch a modal listing available java versions (in a table so you can see path + java version). browse should open up a file explorer. test should test the JRE (doesn't have to have a response to user input for now, when we have a notifs system it should)

Reply:
Auto Detect and Browse are now wired up. Wyatt is working on code for testing the JRE.